tsdb/engine/tsm1/cache.go

```diff
@@ -180,39 +187,49 @@ func (c *Cache) Keys() []string {
 
 // Values returns a copy of all values, deduped and sorted, for the given key.
 func (c *Cache) Values(key string) Values {
-	values, needSort := func() (Values, bool) {
-		c.mu.RLock()
-		defer c.mu.RUnlock()
-		e := c.store[key]
+	c.mu.Lock()
```

Comment:
It's possible that I could limit the amount of time the RW lock is held, to just the time it takes to get a copy of the hot data. Then take the RLock when merging with the snapshots. I need to think about this more.